packages/main/src/Form.ts

```diff
@@ -0,0 +1,437 @@
+	 * Either use FormGroups and make sure all FormItems are part of a FormGroup, or use just FormItems without any FormGroups.
+	 * @public
+	 */
+	@slot({
+		type: HTMLElement,
+		"default": true,
+		individualSlots: true,
+		invalidateOnChildChange: true,
+	})
+	items!: Array<FormItem | FormGroup>;
+
+	/**
+	 * @private
+	 */
+	@property({ validator: Integer, defaultValue: 1 })
+	columnsS!: number;
+	@property({ validator: Integer, defaultValue: 12 })
+	labelSpanS!: number;
+
+	@property({ validator: Integer, defaultValue: 1 })
+	columnsM!: number;
+	@property({ validator: Integer, defaultValue: 4 })
+	labelSpanM!: number;
+
+	@property({ validator: Integer, defaultValue: 2 })
+	columnsL!: number;
+	@property({ validator: Integer, defaultValue: 4 })
+	labelSpanL!: number;
+
+	@property({ validator: Integer, defaultValue: 2 })
+	columnsXl!: number;
+	@property({ validator: Integer, defaultValue: 4 })
+	labelSpanXl!: number;
+
+	onBeforeRendering() {
+		// Parse the layout and set it to the FormGroups/FormItems.
+		this.setColumnLayout();
+
+		// Parse the labelSpan and set it to the FormGroups/FormItems.
+		this.setLabelSpan();
+
+		// Define how many columns a group should take.
+		this.setGroupsColSpan();
+
+		// Create additional CSS for number of columns that are not supported by default.
+		this.createStepCSSStyleSheet();
+	}
+
+	setColumnLayout() {
+		const layoutArr = this.layout.split(" ");
+		layoutArr.forEach((breakpont: string) => {
+			if (breakpont.startsWith("S")) {
+				this.columnsS = parseInt(breakpont.slice(1));
+			} else if (breakpont.startsWith("M")) {
+				this.columnsM = parseInt(breakpont.slice(1));
+			} else if (breakpont.startsWith("L")) {
+				this.columnsL = parseInt(breakpont.slice(1));
+			} else if (breakpont.startsWith("XL")) {
+				this.columnsXl = parseInt(breakpont.slice(2));
+			}
+		});
+	}
+
+	setLabelSpan() {
+		this.labelSpan.split(" ").forEach((breakpont: string) => {
+			if (breakpont.startsWith("S")) {
+				this.labelSpanS = parseInt(breakpont.slice(1));
+			} else if (breakpont.startsWith("M")) {
+				this.labelSpanM = parseInt(breakpont.slice(1));
+			} else if (breakpont.startsWith("L")) {
+				this.labelSpanL = parseInt(breakpont.slice(1));
+			} else if (breakpont.startsWith("XL")) {
+				this.labelSpanXl = parseInt(breakpont.slice(2));
+			}
+		});
+
+		this.items.forEach((item: FormItem | FormGroup) => {
+			item.labelSpan = this.labelSpan;
+			item.itemSpacing = this.itemSpacing;
+		});
+	}
+
+	setGroupsColSpan() {
+		if (!this.hasGroupItems) {
+			return;
+		}
+
+		const itemsCount = this.items.length;
+		const sortedItems = [...this.items].sort((itemA: FormGroup | FormItem, itemB: FormGroup | FormItem) => {
+			return (itemB as FormGroup)?.children?.length - (itemA as FormGroup)?.children?.length;
+		});
+
+		sortedItems.forEach((item: FormGroup | FormItem, idx: number) => {
```

Comment:
you can create simple type guard for this one since we have isGroup getter instead of casting item to FormGroup multiple times